### PR TITLE
Changed CertificatePinner to ignore case when matching pins.

### PR DESCRIPTION
**Pull request by dave-r12**
Closes: https://github.com/square/okhttp/issues/2609

### CODE REVIEW

**Comment by swankjesse** [comment]
I wonder if a better strategy is to fully canonicalized the host, possibly by roundtripping it through HttpUrl.

```
HttpUrl.parse("http://" + hostWithoutWildcard). host()
```
That'd catch punycode also.
The

**Comment by dave-r12** [comment]
Gotcha, that does sound like a better strategy. I'll see if I can rework this.

File: okhttp/src/main/java/okhttp3/CertificatePinner.java [hunk 1]
```diff
@@ -237,15 +237,21 @@ static ByteString sha256(X509Certificate x509Certificate) {
...
+    /** The canonical hostname, i.e. {@code EXAMPLE.com} becomes {@code example.com}. */
+    final String canonicalHostname;
     /** Either {@code sha1/} or {@code sha256/}. */
     final String hashAlgorithm;
     /** The hash of the pinned certificate using {@link #hashAlgorithm}. */
     final ByteString hash;
 
     Pin(String pattern, String pin) {
       this.pattern = pattern;
+      this.canonicalHostname = pattern.startsWith(WILDCARD)
```
Comment (swankjesse) [inline okhttp/src/main/java/okhttp3/CertificatePinner.java]: Nice.

**Comment by swankjesse** [comment]
This is very nice. As an added bonus it’ll blow up if the string is bogus.

**Pull request merged**